.github/workflows/python-app.yml

```diff
@@ -105,15 +115,87 @@ jobs:
       run: |
         make lint-type-check
 
-    - name: Run Tests Coverage
+    - name: Run test on Node ${{ matrix.ci_node_index }}
+      run: |
+        set -e
+        mkdir -p test-results
+        rm -r test-results/parts || echo "no files to delete"
+        OUT=$(python -c "from tests.utils import split_tests; split_tests('test-results/results.xml', ${{ matrix.ci_node_total }}, ${{ matrix.ci_node_index }})")
+        IFS=',' read -a FILE_INFO <<< "$OUT"
+        FILES=("${FILE_INFO[@]:1}")
+        echo "Selected Files are: " ${FILES[@]}
+        echo "Previous timings are: ${FILE_INFO[0]}"
+        ./coverage/coverage.sh results-${{ matrix.python }}-${{ matrix.ci_node_index }}.xml ${FILES[@]}
+
+    - name: Upload Artifact(results)
+      uses: actions/upload-artifact@v3
+      with:
+        name: results-${{ matrix.python }}-${{ matrix.ci_node_index }}
```

Comment:
this one is even more different than above

Reply:
This is just for artifact. Once workflow completes, we cannot access artifact using keys. We can use artifacts inside the workflow only.

The key above is just to pass information from `tests` to `coverage`.